relay-threading/src/multiplexing.rs

```diff
@@ -141,6 +152,13 @@ where
         let mut this = self.project();
 
         loop {
+            // We report how many tasks are being concurrently polled in this future.
+            relay_statsd::metric!(
+                gauge(AsyncPoolGauges::AsyncPoolFuturesPerThread) = this.tasks.len() as u64,
+                pool_name = &this.pool_name,
+                thread_name = &this.thread_name
+            );
```

Comment:
this is probably too high cardinality, maybe we just emit the total size somewhere or the max?

Also we don't have to do it on each poll only when the futures length changes (e.g. on push)